client/my-sites/backup/wpcom-backup-upsell.tsx

```diff
@@ -86,6 +90,76 @@ const BackupUpsellBody = () => {
 	const isJetpack = useSelector( ( state ) => siteId && isJetpackSite( state, siteId ) );
 	const isAtomic = useSelector( ( state ) => siteId && isSiteWpcomAtomic( state, siteId ) );
 	const isWPcomSite = ! isJetpack || isAtomic;
+
+	const [ isRevertedWithValidBackup, setIsRevertedWithValidBackup ] = useState< null | boolean >(
+		null
+	);
+
+	const [ rewindId, setRewindId ] = useState< string | null >( null );
+	const [ backupPeriodDate, setBackupPeriodDate ] = useState< string | null >( null );
+
+	const fetchLatestAtomicTransfer = useCallback( async ( siteId: number | string ) => {
+		try {
+			const transfer = await wpcom.req.get( {
+				path: `/sites/${ siteId }/atomic/transfers/latest`,
+				apiNamespace: 'wpcom/v2',
+			} );
+			return transfer;
+		} catch ( error ) {
+			return null;
+		}
+	}, [] );
+
+	const fetchRewindBackups = useCallback( async ( siteId: number | string ) => {
+		try {
+			const response = await wpcom.req.get( {
+				path: `/sites/${ siteId }/rewind/backups`,
+				apiNamespace: 'wpcom/v3',
+				query: { number: 10 },
+			} );
+
+			const validBackup = response.backups?.find(
+				( backup: { is_rewindable: boolean; summary: string } ) =>
+					backup.is_rewindable && backup.summary === 'Backup complete'
+			);
+
+			if ( validBackup ) {
+				const backupPeriod = validBackup.object.backup_period;
+				const backupPeriodHumanReadable = new Date(
+					parseInt( backupPeriod ) * 1000
+				).toLocaleDateString( 'en-US', { year: 'numeric', month: 'long', day: 'numeric' } );
+
+				setIsRevertedWithValidBackup( true );
+				setRewindId( validBackup.object.backup_period );
+				setBackupPeriodDate( backupPeriodHumanReadable );
+			} else {
+				setIsRevertedWithValidBackup( false );
+				setRewindId( null );
+				setBackupPeriodDate( null );
+			}
+		} catch ( error ) {
+			setIsRevertedWithValidBackup( false );
+			setRewindId( null );
+			setBackupPeriodDate( null );
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( siteId ) {
+			( async () => {
+				const transferStatus = await fetchLatestAtomicTransfer( siteId );
+
+				if ( transferStatus && transferStatus.status === 'reverted' ) {
+					await fetchRewindBackups( siteId );
+				} else {
+					setIsRevertedWithValidBackup( false );
+					setRewindId( null );
+					setBackupPeriodDate( null );
```

Comment:
I think it would be nice to create a helper function to reset these states, to prevent duplicated code.

Something like:

```
const resetBackupState = () => {
  setIsRevertedWithValidBackup( false );
  setRewindId( null );
  setBackupPeriodDate( null );
};
```

And then you can use it on the multiple instances you need to reset the state.

Reply:
Yes...I see those three states were being set three different times  🤦🏻‍♂️

Fixed!